Bottom: bebe23cadb8d11e694b190822e036b1e457abb4f
Top:    ecefe8e4153d53bf6f9328358cbb404847a0dc20
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-25 13:15:45 +0000

Fixup Z-heights of exported copper


---

diff --git a/src/hid/common/object3d.c b/src/hid/common/object3d.c
index ac52b12..66f2528 100644
--- a/src/hid/common/object3d.c
+++ b/src/hid/common/object3d.c
@@ -694,14 +694,14 @@ object3d_from_board_outline (void)
 
 #ifdef REVERSED_PCB_CONTOURS
   objects = object3d_from_contours (board_outline,
-                                    -HACK_BOARD_THICKNESS + HACK_MASK_THICKNESS, /* Bottom */
-                                    0                     -HACK_MASK_THICKNESS,  /* Top */
+                                    -HACK_BOARD_THICKNESS, /* Bottom */
+                                    0                    ,  /* Top */
                                     board_appearance,
                                     top_bot_appearance);
 #else
   objects = object3d_from_contours (board_outline,
-                                     HACK_BOARD_THICKNESS / 2 - HACK_MASK_THICKNESS, /* Bottom */
-                                    -HACK_BOARD_THICKNESS / 2 + HACK_MASK_THICKNESS, /* Top */
+                                     HACK_BOARD_THICKNESS / 2, /* Bottom */
+                                    -HACK_BOARD_THICKNESS / 2, /* Top */
                                     board_appearance,
                                     top_bot_appearance);
 #endif
@@ -961,14 +961,14 @@ object3d_from_soldermask_within_area (POLYAREA *area, int side)
 
 #ifdef REVERSED_PCB_CONTOURS
   objects = object3d_from_contours (info.poly,
-                                    (side == TOP_SIDE) ? -HACK_MASK_THICKNESS : -HACK_BOARD_THICKNESS,                       /* Bottom */
-                                    (side == TOP_SIDE) ? 0                    : -HACK_BOARD_THICKNESS + HACK_MASK_THICKNESS, /* Top */
+                                    (side == TOP_SIDE) ? 0                   - HACK_COPPER_THICKNESS : -HACK_BOARD_THICKNESS - HACK_COPPER_THICKNESS - HACK_MASK_THICKNESS, /* Bottom */
+                                    (side == TOP_SIDE) ? HACK_MASK_THICKNESS - HACK_COPPER_THICKNESS : -HACK_BOARD_THICKNESS - HACK_COPPER_THICKNESS,                       /* Top */
                                     mask_appearance,
                                     NULL);
 #else
   objects = object3d_from_contours (info.poly,
-                                    (side == TOP_SIDE) ? -HACK_BOARD_THICKNESS / 2 + HACK_MASK_THICKNESS : HACK_BOARD_THICKNESS / 2,                       /* Bottom */
-                                    (side == TOP_SIDE) ? -HACK_BOARD_THICKNESS / 2                       : HACK_BOARD_THICKNESS / 2 - HACK_MASK_THICKNESS, /* Top */
+                                    (side == TOP_SIDE) ? -HACK_BOARD_THICKNESS / 2 - HACK_COPPER_THICKNESS                       : HACK_BOARD_THICKNESS / 2 + HACK_COPPER_THICKNESS + HACK_MASK_THICKNESS, /* Bottom */
+                                    (side == TOP_SIDE) ? -HACK_BOARD_THICKNESS / 2 - HACK_COPPER_THICKNESS - HACK_MASK_THICKNESS : HACK_BOARD_THICKNESS / 2 + HACK_COPPER_THICKNESS, /* Top */
                                     mask_appearance,
                                     NULL);
 #endif
@@ -1333,7 +1333,7 @@ compute_depth (int group)
   int min_copper_group;
   int max_copper_group;
   int num_copper_groups;
-  int middle_copper_group;
+//  int middle_copper_group;
   int depth;
 
   top_group = GetLayerGroupNumberBySide (TOP_SIDE);
@@ -1342,12 +1342,12 @@ compute_depth (int group)
   min_copper_group = MIN (bottom_group, top_group);
   max_copper_group = MAX (bottom_group, top_group);
   num_copper_groups = max_copper_group - min_copper_group;// + 1;
-  middle_copper_group = min_copper_group + num_copper_groups / 2;
+//  middle_copper_group = min_copper_group + num_copper_groups / 2;
 
   if (group >= 0 && group < max_group) {
     if (group >= min_copper_group && group <= max_copper_group) {
       /* XXX: IS THIS INCORRECT FOR REVERSED GROUP ORDERINGS? */
-      depth = -(group - middle_copper_group) * HACK_BOARD_THICKNESS / num_copper_groups;
+      depth = -(group - min_copper_group) * (HACK_BOARD_THICKNESS + HACK_COPPER_THICKNESS) / num_copper_groups;
     } else {
       depth = 0;
     }
@@ -1533,7 +1533,11 @@ object3d_from_copper_layers_within_area (POLYAREA *area)
 
   for (group = min_phys_group; group <= max_phys_group; group++) {
 
-    Coord depth = compute_depth (group);
+#ifdef REVERSED_PCB_CONTOURS
+    Coord depth = compute_depth (group) - HACK_BOARD_THICKNESS;
+#else
+    Coord depth = compute_depth (group) + HACK_BOARD_THICKNESS / 2;
+#endif
     info.poly = NULL;
 
     fprintf (stderr, "Computing copper geometry for group %i\n", group);
@@ -1610,8 +1614,8 @@ object3d_from_copper_layers_within_area (POLYAREA *area)
                               depth,                         /* Bottom */
                               depth + HACK_COPPER_THICKNESS, /* Top */
 #else
-                              -depth + HACK_COPPER_THICKNESS, /* Bottom */
-                              -depth,                         /* Top */
+                              -depth,                         /* Bottom */
+                              -depth - HACK_COPPER_THICKNESS, /* Top */
 #endif
                               copper_appearance,
                               NULL));
